src/query/catalog/src/table_context.rs

```diff
@@ -141,7 +141,7 @@ pub trait TableContext: Send + Sync {
         catalog_name: &str,
         database_name: &str,
         table_name: &str,
-        files: Vec<StageFileInfo>,
+        mut files: Vec<StageFileInfo>,
```

Comment:
```suggestion
        files: Vec<StageFileInfo>,
```
please don't make it to mut.